src/Interactive/Host/Interactive/Core/InteractiveHost.cs

```diff
@@ -175,36 +152,23 @@ private RemoteService TryStartProcess(string hostPath, CultureInfo culture, Canc
                     newProcessId = 0;
                 }
 
-                // sync:
-                while (!semaphore.WaitOne(_millisecondsTimeout))
-                {
-                    if (!CheckAlive(newProcess, hostPath))
-                    {
-                        return null;
-                    }
-
-                    WriteOutputInBackground(isError: false, string.Format(InteractiveHostResources.Attempt_to_connect_to_process_Sharp_0_failed_retrying, newProcessId));
-                    cancellationToken.ThrowIfCancellationRequested();
-                }
+                // (miziga) initializing pipe for replacement of remote channel
+                NamedPipeClientStream _clientStream = new NamedPipeClientStream(".", pipeName, PipeDirection.InOut, PipeOptions.Asynchronous);
+                JsonRpc jsonRpc = JsonRpc.Attach(_clientStream);
 
                 // instantiate remote service:
-                Service newService;
                 try
                 {
-                    newService = (Service)Activator.GetObject(
-                        typeof(Service),
-                        "ipc://" + remoteServerPort + "/" + Service.ServiceName);
-
                     cancellationToken.ThrowIfCancellationRequested();
-
-                    newService.Initialize(_replServiceProviderType, culture.Name);
+                    await _clientStream.ConnectAsync().ConfigureAwait(false);
+                    await jsonRpc.InvokeAsync<Task>("Initialize", _replServiceProviderType, culture.Name).ConfigureAwait(false);
                 }
                 catch (RemotingException) when (!CheckAlive(newProcess, hostPath))
```

Comment:
>RemotingException [](start = 23, length = 17)

This exception type is from the legacy remoting framework we are replacing. See comments on InvokeAsync to the list of exceptions it may throw. We want to catch ObjectDisposedException here with the !CheckAlive condition. The connection may be disposed if the process is terminated for some reason, so that's what we are trying to handle.

RemoteInvocationException shouldn't happen - that would mean the Initialize method threw an exception on the server side.